src/audio/mod.rs

```diff
@@ -61,21 +61,54 @@ pub fn run(
     Ok(())
 }
 
-pub fn devices() -> Result<(Device, Device)> {
+pub fn devices(
+    input_device: &Option<String>,
+    output_device: &Option<String>,
+) -> Result<(Device, Device)> {
     let host = cpal::default_host();
-    let input_device = host
-        .default_input_device()
-        .ok_or_else(|| anyhow!("Failed to find input device"))?;
+
+    let input_device = match input_device {
+        Some(name) => device(&host, &name),
+        None => host
+            .default_input_device()
+            .ok_or_else(|| anyhow!("Failed to find input device")),
+    }?;
```

Comment:
If a name is provided in the config file, use it, otherwise, get the default device.

Reply:
It seems like macOS won’t allow configurations that it doesn’t agree with, but they just fail at runtime in weird ways. I couldn’t figure out how to distinguish between input/output devices without actually trying to start them up. 🤷🏻